### PR TITLE
Add witness inclusion and bad rssi witness checks

### PR DESCRIPTION
**Pull request by vihu**
- Add witness filter for impossible RSSI values
- Add witness filter for witness who aren't in ActiveGateways

### CODE REVIEW

File: src/blockchain_poc_path_v2.erl [hunk 1]
```diff
@@ -305,6 +307,45 @@ check_witness_distance(WitnessParent, ParentIndices, ExclusionCells) ->
...
                   end, ParentIndices)).
 
+-spec check_witness_bad_rssi(Witness :: blockchain_ledger_gateway_v2:gateway_witness(),
+                             Vars :: map()) -> boolean().
+check_witness_bad_rssi(Witness, Vars) ->
+    try
+        blockchain_ledger_gateway_v2:witness_hist(Witness)
+    of
+        Hist ->
+            case poc_version(Vars) of
```
Comment (Vagabond) [inline src/blockchain_poc_path_v2.erl]: Check this first before pulling the histogram?

Comment (vihu) [reply src/blockchain_poc_path_v2.erl]: Check: 6ae8e49

**Comment by vihu** [comment]
Do not merge, couldn't resync miner with this branch. Looking.

File: src/blockchain_poc_path_v2.erl [hunk 2]
```diff
@@ -305,6 +307,45 @@ check_witness_distance(WitnessParent, ParentIndices, ExclusionCells) ->
                           h3:grid_distance(WitnessParent, ParentIndex) < ExclusionCells
                   end, ParentIndices)).
 
+-spec check_witness_bad_rssi(Witness :: blockchain_ledger_gateway_v2:gateway_witness(),
+                             Vars :: map()) -> boolean().
+check_witness_bad_rssi(Witness, Vars) ->
+    case poc_version(Vars) of
+        V when V > 4 ->
```
Comment (Vagabond) [inline src/blockchain_poc_path_v2.erl]:
```
1> undefined > 4.
true
```

Comment (Vagabond) [inline src/blockchain_poc_path_v2.erl]:
```suggestion
        V when is_integer(V), V > 4 ->
```

**Comment by vihu** [comment]
Superseded by #314

**Pull request closed without merging**